File: algorithms/1876.substrings-of-size-three-with-distinct-characters/solution.go
func countGoodSubstrings(s string) int {
	var count int

	for i := 2; i < len(s); i++ {
		if s[i-2] != s[i-1] && s[i-1] != s[i] && s[i] != s[i-2] {
			count++
		}
	}

	return count
}

File: algorithms/1876.substrings-of-size-three-with-distinct-characters/solution.rs
impl Solution {
    pub fn count_good_substrings(s: String) -> i32 {
        let mut count = 0;

        let c = s.chars().collect::<Vec<char>>();
        for i in 2..s.len() {
            if c[i - 2] != c[i - 1] && c[i - 1] != c[i] && c[i] != c[i - 2] {
                count += 1;
            }
        }

        count
    }
}
